### ISSUE TITLE
pyright does not catch 'Returning Any from function declared to return "None"'

### ISSUE THREAD

**Issue by he1f** [microsoft/pyright#8212]
**Describe the bug**
pyright in strict mode does not see problems in the following code. At the same time latest version of mypy reports that:
```
C:\dmk\python\qqqq>mypy --strict test.py                                                
test.py:7: error: Returning Any from function declared to return "None"  [no-any-return]
Found 1 error in 1 file (checked 1 source file)                                         
```

**Code or Screenshots**
```python
from typing import Any

def bar() -> Any:
    return True

def foo() -> None:
    return bar()
```

A little bit more complicated example does not catch neither by pyright nor by mypy:

```python
from typing import Tuple, Any

def bar() -> Tuple[int, Any, int]:
    return (1, True, 1)

def foo() -> Tuple[int, None, int]:
    return bar()
```

**VS Code extension or command-line**
2024.6.1 (6 June 2024) Release

**Comment by erictraut** [comment]
Pyright is working correctly here, so this is not a bug. In the Python static type system, the `Any` type is assignable to all other types (and vice versa). For details, refer to [this section of the Python typing spec](https://typing.readthedocs.io/en/latest/spec/concepts.html#the-assignable-to-or-consistent-subtyping-relation). No error should be given in this case.

Pyright and mypy differ in the handling of `Any` in strict modes. Pyright distinguishes between "implicit" and "explicit" `Any`, and it refers to the former as `Unknown`. It never generates errors for explicit `Any`, since those types are explicitly specified. For more details, refer to [this documentation](https://microsoft.github.io/pyright/#/mypy-comparison?id=unknown-type-and-strict-mode).
